src/commodore-helm/index.ts

```diff
@@ -0,0 +1,243 @@
+  config: any
+): PackageFile | null {
+  logger.debug(
+    { fileName, depName: config.depName, baseDeps: config.baseDeps },
+    'extractPackageFile upgrade'
+  );
+  if (!config.depName || !config.baseDeps) {
+    logger.warn(
+      { fileName },
+      'extractPackageFile() called for file without dependency info in config'
+    );
+    return null;
+  }
+  if (path.parse(fileName).name != 'defaults') {
+    logger.warn(
+      { fileName },
+      'extractPackageFile() called for a package file other than `defaults.yml`, returning null'
+    );
+    return null;
+  }
+  const defaults: any = yaml.load(content);
+  if (!defaults || !defaults.parameters) {
+    return null;
+  }
+  const dep = config.baseDeps.find((d: PackageDependency) => {
+    return d.depName === config.depName;
+  });
+  const componentKey: string = componentKeyFromName(dep.groupName);
+  const charts: Map<string, string> = defaults.parameters[componentKey].charts;
+  if (!charts) {
+    return null;
+  }
+
+  let deps: PackageDependency[] = Object.entries(charts).map(
+    ([chartName, chartVersion]) => {
+      let d = config.baseDeps.find((d: PackageDependency) => {
+        return d.propSource == chartName;
+      });
+      let res: PackageDependency = {
+        depName: d.depName,
+        currentValue: chartVersion,
+      };
+      return res;
+    }
+  );
+
+  return { deps };
+}
+
+export async function extractAllPackageFiles(
+  _config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  let fileContents: Map<string, string> = new Map();
+  let componentName: string | null = null;
+  let defaults_file: string | null = null;
+
+  for (const file of packageFiles) {
+    const content = await readLocalFile(file, 'utf8');
+    const key = path.parse(file).name;
+    fileContents.set(key, content);
+    if (key != 'defaults') {
+      componentName = key;
+    } else {
+      defaults_file = file;
+    }
+  }
+
+  if (!fileContents.has('defaults')) {
+    logger.error('Component repository has no `class/defaults.ya?ml`');
+    return null;
+  }
+
+  if (!componentName) {
+    logger.error(
+      { packageFiles },
+      'Unable to identify component name from package files'
+    );
+    return null;
+  }
+
+  const defaults: any = yaml.load(fileContents.get('defaults') ?? '');
+  const component: any = yaml.load(fileContents.get(componentName) ?? '');
+
+  if (
+    defaults === undefined ||
+    defaults === null ||
+    defaults.parameters === undefined ||
+    defaults.parameters === null ||
+    component === undefined ||
+    component === null ||
+    component.parameters === undefined ||
+    component.parameters === null
+  ) {
+    return null;
+  }
+
+  const deps: PackageDependency[] = extractHelmChartDependencies(
+    componentName,
+    defaults.parameters,
+    component.parameters
+  );
+
+  if (deps.length == 0) {
+    return null;
+  }
+
+  return [{ packageFile: defaults_file, datasource: 'helm', deps }];
```

Comment:
```suggestion
  return [{ packageFile: defaults_file as string, datasource: 'helm', deps }];
```
We know `defaults_file` cannot be null, but my language server doesn't.

Reply:
Adjusted by making exit when `defaults_file` is null explicit in https://github.com/projectsyn/commodore-renovate/pull/17/commits/d46d490279cfef14b2b87129908e99311eb707fe